### ISSUE TITLE
[BUG] request-id correlation header not being send because enableCorsCorrelation is false default, set enableCorsCorrelation to true by default

### ISSUE THREAD

**Issue by llatinov** [microsoft/ApplicationInsights-JS#1550]
I have enabled tracking on both the frontend (React app) and the backend (.NET Core app).

When I run it in development mode, I do not use CORS preflight requests (this is really details, not important for the current issue. There is no CORS because I have a React proxy setup and the frontend is hitting the same port on which the proxy works, so the browser does not issue a preflight request).

When I built the frontend and deploy it to real development instance, frontend and backend work on separate domains, so CORS are needed, and the browser issues preflight requests.

I used the default configuration, just enabled the fetch tracking (disableFetchTracking: false) and surprise - while I develop everything works like a charm, and correlation is done. Then I deploy to develop and correlation breaks.

After doing some investigation I found the root cause: https://github.com/microsoft/ApplicationInsights-JS/blob/b63955bbdb47ac24dbb76479309628a47188fe0a/shared/AppInsightsCommon/src/Util.ts#L358

And then: https://github.com/microsoft/ApplicationInsights-JS/blob/b63955bbdb47ac24dbb76479309628a47188fe0a/extensions/applicationinsights-dependencies-js/src/ajax.ts#L279

So basically because enableCorsCorrelation is false, then no correlation header is included. I believe this is a problem.

I would say:
1. enableCorsCorrelation should be true by default
2. The if condition in canIncludeCorrelationHeader should be evaluated and may be reworked

**Comment by thoo1** [comment]
I would say `enableCorsCorrelation` is default false because the library cannot know if sending the correlation headers will fail CORS (server may reject the OPTIONS preflight)

Our app has this same issue where we turned `enableCorsCorrelation: true`. We saw some requests fail because the service would reject the request as the request was trying to send the correlation headers. We're moving from a "deny-list" (dont send correlation headers for x,y,z domains) to an allow-list (only send correlation headers for a,b,c domains where it is known that CORS will succeed)

Hope this helps

**Comment by MSNev** [comment]
For us, changing defaults like this would be considered a breaking change and therefore require a major version bump, so if we do this it won't be until then.
